src/vm_cpu.rs

```diff
@@ -62,11 +65,22 @@ struct HostCpuState {
 struct GuestCpuState {
     gprs: GeneralPurposeRegisters,
     fprs: FloatingPointRegisters,
+    #[cfg(feature = "vectors")]
+    vprs: VectorRegisters,
     fcsr: u64,
     sstatus: u64,
     hstatus: u64,
     scounteren: u64,
     sepc: u64,
+
+    #[cfg(feature = "vectors")]
+    vstart: u64,
+    #[cfg(feature = "vectors")]
+    vcsr: u64,
+    #[cfg(feature = "vectors")]
+    vtype: u64,
+    #[cfg(feature = "vectors")]
+    vl: u64,
```

Comment:
Can you put all the vector state in its own struct and then wrap that in a `#[cfg]` instead of repeating it a whole bunch of times here? Something like:

```
#[cfg(...)}
struct VectorState {
   regs: VectorRegisters,
   ...
}
#[cfg(not(...))]
struct VectorState {}
```

Technically you don't even need to put the struct behind a `#[cfg]` since it'll compile just fine, though at the expense of extra storage when vectors are disabled.

Reply:
fixed